src/main/webapp/app/course/dashboards/assessment-dashboard/exam-assessment-buttons/exam-assessment-buttons.component.ts

```diff
@@ -64,18 +64,24 @@ export class ExamAssessmentButtonsComponent implements OnInit {
             this.courseService.find(this.courseId).subscribe((courseResponse) => {
                 this.course = courseResponse.body!;
             });
-            const studentExamObservable = this.studentExamService.findAllForExam(this.courseId, this.examId).pipe(
-                tap((res) => {
-                    // TODO: directly load the longest working time from the server and avoid loading all student exams
-                    this.setStudentExams(res.body);
-                    this.longestWorkingTime = Math.max.apply(
-                        null,
-                        this.studentExams.map((studentExam) => studentExam.workingTime),
-                    );
+
+            /*
+            Prepare workingTimeObservable to perform the following on subscribe:
+            - set the longestWorkingTime
+            - trigger (re)calculation of whether the exam is over
+             */
+            const workingTimeObservable = this.studentExamService.getLongestWorkingTimeForExam(this.courseId, this.examId).pipe(
+                tap((value) => {
+                    this.longestWorkingTime = value;
                     this.calculateIsExamOver();
                 }),
             );
 
+            /*
+            Prepare workingTimeObservable to perform the following on subscribe:
+            - set the exam
+            - trigger (re)calculation of whether the exam is over
+             */
```

Comment:
I think there is a mistake in the comment, I would be in favor of removing the comment

```suggestion
            /*
            Prepare examObservable to perform the following on subscribe:
            - set the exam
            - trigger (re)calculation of whether the exam is over
             */
```